File: ios/Plugin/OpenWithPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(OpenWithPlugin, "OpenWith",
   CAP_PLUGIN_METHOD(openWith, CAPPluginReturnPromise);
)

File: ios/Plugin/OpenWithPlugin.swift
import Foundation
import Capacitor

@objc(OpenWithPlugin)
public class OpenWithPlugin: CAPPlugin {

    let openWith = OpenWith()

    @objc func openWith(_ call: CAPPluginCall) {
        guard let path = call.getString("path") else {
            call.reject("Path is required")
            return
        }

        DispatchQueue.main.async {
            if let viewController = self.bridge?.viewController {
                self.openWith.openFile(path: path, viewController: viewController)
                call.resolve(["message": "File opened successfully"])
            } else {
                call.reject("Failed to open file")
            }
        }
    }
}

